packages/calypso-e2e/src/lib/blocks/block-flows/layout-grid.ts

```diff
@@ -115,14 +115,12 @@ export class LayoutGridBlockFlow implements BlockFlow {
 	 * @param {PublishedPostContext} context The current context for the published post at the point of test execution.
 	 */
 	async validateAfterPublish( context: PublishedPostContext ): Promise< void > {
-		const expectedLeftColumnTextLocator = context.page.locator(
-			`text=${ this.configurationData.leftColumnText }`
-		);
-		await expectedLeftColumnTextLocator.waitFor();
+		await context.page
+			.locator( `:text("${ this.configurationData.leftColumnText }"):visible` )
+			.waitFor();
 
-		const expectedRightColumnTextLocator = context.page.locator(
-			`text=${ this.configurationData.rightColumnText }`
-		);
-		await expectedRightColumnTextLocator.waitFor();
+		await context.page
+			.locator( `:text("${ this.configurationData.rightColumnText }"):visible` )
+			.waitFor();
```

Comment:
The previous selector ended up matching on more than 1 element on the page, but one of the element was always hidden.

By specifying the `:visible` pseudo-CSS class, we are able to limit the match to visible elements.